src/main/java/com/google/sps/data/Post.java

```diff
@@ -212,13 +260,30 @@ public Entity postToEntity(String entityKind) {
     newPost.setProperty("typeOfFood", typeOfFood);
     newPost.setProperty("numberOfPeopleItFeeds", numberOfPeopleItFeeds);
     newPost.setProperty("description", description);
+    newPost.setProperty("rank", rank);
 
     newPost.setProperty("timeSort", timeSort);
     newPost.setProperty("collegeId", collegeId);
 
     return newPost;
   }
 
+  /* Set sorting by the rank of event. */
+  @Override     
+  public int compareTo(Post post) {          
+    return (this.getRank() > post.getRank() ? 1 : 
+      (this.getRank() == post.getRank() ? 0 : -1));     
```

Comment:
`return this.getRank() - post.getRank()`. 

`compareTo` does not need to be one of `{0, -1, 1}` so, negative/positive/0 is sufficient.

Reply:
Gotcha!